src/Migrate.php

```diff
@@ -0,0 +1,1394 @@
+                    if ($this->config->getSynchronizeRun()) {
+                        $this->grantsPrivilegesToOldDatabase($database, $rolesGrant['name'], $mainRole);
+                    }
+                }
+            }
+
+            foreach ($userGrants as $userGrant) {
+                $this->createUser($userGrant, $this->config->getPasswordOfUsers());
+            }
+        }
+
+        $this->logger->info('Migrating grants of warehouses/users and roles.');
+        // second step - migrate all grants of roles/users/warehouses/account
+        foreach ($this->databases as $database) {
+            $databaseRole = $this->sourceConnection->getOwnershipRoleOnDatabase($database);
+
+            if ($this->config->getSynchronizeRun()) {
+                $this->grantsPrivilegesToOldDatabase($database, $databaseRole, $mainRole);
+            }
+
+            [
+                'grants' => [
+                    'roles' => $rolesGrants,
+                    'warehouse' => $warehouseGrants,
+                    'user' => $userGrants,
+                ],
+            ] = Helper::parseGrantsToObjects($grants[$databaseRole]);
+
+            foreach ($rolesGrants as $rolesGrant) {
+                $this->destinationConnection->assignGrantToRole($rolesGrant);
+            }
+
+            foreach ($warehouseGrants as $warehouseGrant) {
+                $this->destinationConnection->assignGrantToRole($warehouseGrant);
+            }
+
+            foreach ($userGrants as $userGrant) {
+                $this->destinationConnection->assignGrantToRole($userGrant);
+            }
+        }
+    }
+
+    public function cloneDatabaseWithGrants(string $mainRole, array $grants): void
+    {
+        foreach ($this->databases as $database) {
+            $databaseRole = $this->sourceConnection->getOwnershipRoleOnDatabase($database);
+            [
+                'grants' => [
+                    'databases' => $databaseGrants,
+                    'schemas' => $schemasGrants,
+                    'tables' => $tablesGrants,
+                ],
+                'futureGrants' => [
+                    'tables' => $tablesFutureGrants,
+                ],
+            ] = Helper::parseGrantsToObjects($grants[$databaseRole]);
+
+            $this->destinationConnection->useRole($mainRole);
+
+            $shareDbName = $database . '_SHARE';
+            $oldDbName = $database . '_OLD';
+
+            $sourceDatabases = $this->sourceConnection->fetchAll(sprintf(
+                'SHOW DATABASES LIKE %s',
+                QueryBuilder::quote($database)
+            ));
+            assert(count($sourceDatabases) === 1);
+            $sourceDatabase = current($sourceDatabases);
+
+            $this->logger->info(sprintf('Migrate database "%s".', $database));
+            $this->destinationConnection->query(sprintf(
+                'CREATE DATABASE %s DATA_RETENTION_TIME_IN_DAYS=%s;',
+                Helper::quoteIdentifier($database),
+                '1'
```

Comment:
Proč je tady 1 a ne skutečná hodnota?

Reply:
určitě - tady je https://github.com/keboola/project-migration-tool/pull/1/commits/9128ea55e09b8dc8a8a7ce570453a6a2daf497fe